src/wallet/db.cpp

```diff
@@ -34,10 +27,10 @@ std::vector<fs::path> ListDatabases(const fs::path& wallet_dir)
         const fs::path path = it->path().string().substr(offset);
 
         if (it->status().type() == fs::directory_file &&
-            (ExistsBerkeleyDatabase(it->path()) || ExistsSQLiteDatabase(it->path()))) {
+            (IsBDBFile(BDBDataFile(it->path())) || IsSQLiteFile(SQLiteDataFile(it->path())))) {
```

Comment:
9d1fd52952cb6a944e7de4dd7fca241e3341894b

As a next step it would be nice to refactor this and `MakeWalletDatabase` to avoid duplicate code. Something like:
```cpp
DatabaseStatus CheckWalletDatabase(const fs::path& path);
DatabaseStatus CheckWalletDatabase(const std::string& name)
{
    return CheckWalletDatabase(fs::absolute(name, GetWalletDir()));
}
```
Which would be called in both places.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/20275#discussion_r515641690

> [9d1fd52](https://github.com/bitcoin/bitcoin/commit/9d1fd52952cb6a944e7de4dd7fca241e3341894b)
> 
> As a next step it would be nice to refactor this and `MakeWalletDatabase` to avoid duplicate code.

Definitely could be a next step. I'd be a little skeptical of adding another similar sounding function when it's nice to have fewer distinct functions, but would have to see what this simplifies